### PR TITLE
Send AppLifecycleState notifications on FlutterViewController lifecycle events.

### PR DESCRIPTION
**Pull request by szakarias**
plus small bugfix in engine.cc.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -273,11 +273,17 @@ - (void)loadView {
...
 
+#pragma mark - UIViewController lifecycle notifications
+
 - (void)viewWillAppear:(BOOL)animated {
   [self connectToEngineAndLoad];
   [super viewWillAppear:animated];
 }
 
+- (void)viewDidDisappear:(BOOL)animated {
+  [_lifecycleChannel.get() sendMessage:@"AppLifecycleState.paused"];
```
Comment (mravn-google) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Add super call?

**Comment by Hixie** [comment]
Tests?

**Comment by Hixie** [comment]
cc @cbracken

**Comment by cbracken** [comment]
LGTM for the change. This reveals a gap in our testing. We've got lifecycle state change tests on the framework side (packages/flutter/test/widgets/binding_test.dart) but since we effectively fake out the framework messages in some of those tests, we're not getting integration testing. On iOS I'm not sure how do-able integration testing would be for some of these without some trickery, I assume Android might have a few more options for including app switching via adb?

**Comment by szakarias** [comment]
I am working on an example that uses this, and will also look at some automated testing later.

**Pull request merged**